### DISCUSSION TITLE
DB/db6/

### DISCUSSION THREAD

**Discussion by giscus** [HobbitQia/notebook#28] [General]
# DB/db6/

HobbitQia的笔记本

https://note.hobbitqia.cc/DB/db6/

**Comment by miraitowaves** [comment]
为什么多对多的关系一定要转化成一个关系模式？
我的理解是这样的：
1. 在多对多的关系中，我们无法仅通过在一个表中添加一个引用另一个表主键的外键来实现
2. 如果我们有两个实体，学生和课程。一个学生可以选多门课，一门课可以被多个学生选。在这种情况下，我们无法只添加一个字段在学生表或者课程表来表示这种关系，因为这样的设计无法反映出一个学生可以选多门课程，或者一门课程可以被多个学生选择这样的多对多关系
所以，需要使用额外的表来链接参与多对多关系的两个表
